workflow/workflow_options.go

```diff
@@ -42,9 +42,13 @@ func WithWorkflowTaskList(ctx Context, name string) Context {
 	return internal.WithWorkflowTaskList(ctx, name)
 }
 
-// WithWorkflowTaskListMapper returns a copy of Context with changed tasklist
-func WithWorkflowTaskListMapper(ctx Context, mapper func(name string) string) Context {
-	return internal.WithWorkflowTaskListMapper(ctx, mapper)
+// GetWorkflowTaskList returns a copy of Context with changed tasklist
+// 1. Activity context returns the current workflow tasklist
+// 2. Child workflow context returns the child workflow tasklist
+// 3. Workflow context will return current workflow tasklist
+// 4. If Context is not a workflow context or activity context, it will return nil
```

Comment:
2 things I think:
- indent the list so it renders in docs stuff as a list instead of one giant paragraph
- activity context is just plain wrong, it's different info (though it inherits). 

it's also just kinda generally confusing I think, nil-ness and what it returns depends on if it has been _modified_, right?  e.g. an un-changed tasklist will be nil and implies child workflows and activities will use the `GetInfo(ctx).TaskListName`

Reply:
Should I always return a tasklist then?

Reply:
I've kinda been internally debating about that tbh.  allowing empty/nil is more precise and lets you tell the difference between "this is falling back to the workflow's tasklist" and "this tasklist has been explicitly set", so in a number of ways I think it may actually be better.

Why anyone would _care_ about that distinction, I dunno.  And it does make things more verbose for the likely-very-common case of using the fallback.  I just tend to lean most-precise first, and add other APIs for convenience if it really seems necessary.

Reply:
make sense. 
For `activity context is just plain wrong, it's different info (though it inherits).` Yes, but if user passes a activity context in GetWorkflowTaskList, it still returns the workflow context and thus should be intuitive.